### ISSUE TITLE
Undefined property: ApiPlatform\Hydra\Serializer\CollectionNormalizer::$resourceMetadataCollectionFactory

### ISSUE THREAD

**Issue by Aerendir** [api-platform/core#5787]
**API Platform version(s) affected**: 3.1.15

**Description**  

> Undefined property: ApiPlatform\Hydra\Serializer\CollectionNormalizer::$resourceMetadataCollectionFactor

**How to reproduce**  

...

**Possible Solution**  

Correctly set the dependency as a property.

https://github.com/api-platform/core/blob/0b51d280a101e9f8bd9cfe9c18968164d8e19375/src/Hydra/Serializer/CollectionNormalizer.php#L42

```diff
public function __construct(
        private readonly ContextBuilderInterface $contextBuilder, 
        ResourceClassResolverInterface $resourceClassResolver, 
        private readonly IriConverterInterface $iriConverter, 
-       readonly ?ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null,
+       private readonly ?ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null,
        array $defaultContext = []
    ) {
```

Or, alternatively:

```diff
public function __construct(private readonly ContextBuilderInterface $contextBuilder, ResourceClassResolverInterface $resourceClassResolver, private readonly IriConverterInterface $iriConverter, readonly ?ResourceMetadataCollectionFactoryInterface $resourceMetadataCollectionFactory = null, array $defaultContext = [])
    {
        $this->defaultContext = array_merge($this->defaultContext, $defaultContext);

-        if ($this->resourceMetadataCollectionFactory) {
+        if ($resourceMetadataCollectionFactory) {
            trigger_deprecation('api-platform/core', '3.0', sprintf('Injecting "%s" within "%s" is not needed anymore and this dependency will be removed in 4.0.', ResourceMetadataCollectionFactoryInterface::class, self::class));
        }

        parent::__construct($resourceClassResolver, '');
    }
```

**Additional Context**  

It seems the code was fixed, but a new version was not yet released.

So, basically, the version `3.1.15` is not usable, but no version `3.1.16` was published: am I right? Any idea about when the `3.1.16` release will be published?

**Comment by Aerendir** [comment]
Sorry, it seems was fixed by https://github.com/api-platform/core/pull/5776
